plutus-core/plutus-core/src/PlutusCore/Default/Universe.hs

```diff
@@ -530,9 +532,19 @@ outOfBoundsErr x branches = fold
     ]
 
 instance AnnotateCaseBuiltin DefaultUni where
-    annotateCaseBuiltin (SomeTypeIn uni) branches = case uni of
+    annotateCaseBuiltin ann (SomeTypeIn uni) branches = case uni of
         DefaultUniBool    -> Right $ map (, []) branches
         DefaultUniInteger -> Right $ map (, []) branches
+        DefaultUniList ty ->
+          case branches of
+            [nil, cons] ->
+              Right
+                [ (nil, [])
+                , (cons, [ TyBuiltin @TyName ann $ SomeTypeIn ty
+                         , TyBuiltin @TyName ann $ SomeTypeIn $ DefaultUniList ty
+                         ])
+                ]
+            _ -> Left $ "Casing on list requires two branches"
```

Comment:
I discussed it with @zliu41 and IIRC we agreed that we should allow only one branch too for the cons case, since a lot of time it's expected that the list is gonna be non-empty and it would be wasteful to have a branch for the nil case when it's supposed to fail evaluation anyway.

@zliu41 we still need to reconcile it with `False` being the default for `Bool`. It's not too late to change things I think. Maybe we should make `True` and `(:)` the default branches for consistency. We can do it with or without reordering them. So let's discuss it some time.

Reply:
I added single case as well